### PR TITLE
new commit/read for notebook

### PR DESCRIPTION
**Pull request by songole**

### CODE REVIEW

File: dkube/sdk/api.py [hunk 1]
```diff
@@ -676,21 +678,100 @@ def delete_featuresets(self, featureset_list):
...
+            A dictionary with response status and the deleted featureset name
+
+        """
+        assert(
+            name
+            and isinstance(name, str)
+        ), "Invalid parameter, value must be a featureset name"
+        list = []
+        list.append(name)
+        return super().delete_featureset(list)
```
Comment (mak-454) [inline dkube/sdk/api.py:703]: can do          return super().delete_featureset([name]) 
So the base func takes a list but func here takes only single input?

File: dkube/sdk/internal/api_base.py [hunk 2]
```diff
@@ -133,30 +135,45 @@ def trigger_runs(self, condition):
...
+        return response.to_dict()
+
+    def read_featureset(self, name, version=None, path=None):
+        # Todo: read even if not mounted
+        return DKubeFeatureSetUtils().features_read(name, version, path)
 
     def delete_featureset(self, delete_list):
         response = self._api.featureset_delete({'featuresets': delete_list})
-        return response
+        print(response)
```
Comment (mak-454) [inline dkube/sdk/internal/api_base.py:159]: remove this print?

File: dkube/sdk/rsrcs/featureset.py [hunk 3]
```diff
@@ -70,97 +72,35 @@ def update_features_path(self, path=None):
...
 
         """
-        # Get featurespec from DKube
-        authToken = os.getenv('DKUBE_USER_ACCESS_TOKEN')
-        d_api = api.DkubeApi(token=authToken)
-        res = d_api.get_featurespec(featureset=self.featureset.name)
-        f_spec = res.data
-        if f_spec == None:
-            return {"status": -1, "error": "Featurespec is not found"}
+        if featurespec is None or dataframe is None:
```
Comment (mak-454) [inline dkube/sdk/rsrcs/featureset.py:92]: if both of them are mandatory then in func definition should even make them like optional args.. i mean KWARGS..

File: dkube/sdk/rsrcs/featureset.py [hunk 4]
```diff
@@ -175,11 +115,155 @@ def validate(self, dataframe=None):
         # - The feature schema should be the same.
 
         if len(fspec_keys) != len(df_keys):
```
Comment (mak-454) [inline dkube/sdk/rsrcs/featureset.py:117]: trying to compare for two dicts to be same?
we can just use "==" correct?

File: dkube/sdk/rsrcs/featureset.py [hunk 5]
```diff
@@ -175,11 +115,155 @@ def validate(self, dataframe=None):
...
+        return featureset_metadata
+        # Convert featureset metadata (featurespec) to yaml
+        #featureset_metadata = yaml.dump(featureset_metadata, default_flow_style=False)
+
+    # return the mounted path for the featureset
+    def _get_featureset_mount_path(self, name, config, type):
+        # name - featureset name
+        # config - config.json in dict format
+        # type - search in 'outputs' or 'inputs'
+        for keys in config:
```
Comment (mak-454) [inline dkube/sdk/rsrcs/featureset.py:152]: is config a list ? we could outputs as below?
outputs = config[config.index(type)]

File: dkube/sdk/rsrcs/featureset.py [hunk 6]
```diff
@@ -175,11 +115,155 @@ def validate(self, dataframe=None):
...
+                            for fset in rec[keys]:
+                                if name == fset['name']:
+                                    return fset['location']
+                            return None
+
+    def _get_d3_rel_path(self, full_path):
+        # Get relative path from dkube store                    
+        base = os.getenv("DKUBE_DATA_BASE_PATH")
+        relpath = os.path.relpath(full_path, base)
+        path_list = relpath.split("/")
```
Comment (mak-454) [inline dkube/sdk/rsrcs/featureset.py:167]: can use relpath.replace('home', 'users',1) ?

**Comment by songole** [comment]
@riteshkarvaloc @deepio-oc look at featureset-titanic.py for new API reference

I tested from outside on my vm by faking a notebook environment. Need to test from inside job/notebook.

**Comment by deepio-oc** [comment]
@songole I see many bodyXXX.py files in this PR. I believe this is what you were referring to as missing files. When I generated, these were generated as dataXXX.py and then I had to modify API implementation to pass "data" as the argument in place of body at few places.

I used following command to generate.
```
docker run --rm -v ${PWD}:/local swaggerapi/swagger-codegen-cli generate -i /local/swagger.json -l python -o /local/out/python -DpackageName=dkube_api
```
We should use tagged image. latest is 2.4.18. Could you try generating API stub with that.  if possible, we should separate out the commits (keep generated code in 1 commit) and remaining on top of it

**Comment by songole** [comment]
> @songole I see many bodyXXX.py files in this PR. I believe this is what you were referring to as missing files. When I generated, these were generated as dataXXX.py and then I had to modify API implementation to pass "data" as the argument in place of body at few places.
> 
> I used following command to generate.
> 
> ```
> docker run --rm -v ${PWD}:/local swaggerapi/swagger-codegen-cli generate -i /local/swagger.json -l python -o /local/out/python -DpackageName=dkube_api
> ```
> 
> We should use tagged image. latest is 2.4.18. Could you try generating API stub with that. if possible, we should separate out the commits (keep generated code in 1 commit) and remaining on top of it

I used the same command. I am not able to run featureset.py. Has anything changed?

File: dkube/sdk/rsrcs/featureset.py (hunk 5 above)
Comment (songole) [reply dkube/sdk/rsrcs/featureset.py:152]: config is a dict. it is /etc/dkube/config.json

**Pull request merged**